spring-integration-zeromq/src/main/java/org/springframework/integration/zeromq/dsl/ZeroMq.java

```diff
@@ -58,6 +59,17 @@ public static ZeroMqMessageHandlerSpec outboundChannelAdapter(ZContext context,
 		return outboundChannelAdapter(context, () -> connectUrl);
 	}
 
+	/**
+	 * Create an instance of {@link ZeroMqMessageHandlerSpec} for the provided {@link ZContext} and binding port.
+	 * @param context the {@link ZContext} to use.
+	 * @param port the port to bind ZeroMq socket to over TCP.
+	 * @return the spec.
+	 * @since 6.4
+	 */
+	public static ZeroMqMessageHandlerSpec outboundChannelAdapter(ZContext context, int port) {
```

Comment:
I wonder if DSL has to expose extra factories to explicitly bind to random port.
Something similar what you have with those new ctors in the `ZeroMqMessageHandler`.

Reply:
Or you have missed to push the change, or just didn't implement yet.
Hence the conversion is not resolved yet 😄